azure_functions_worker/mmap_handler/file_accessor.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import abc
+import mmap
+from typing import Optional
+
+
+class FileAccessor(metaclass=abc.ABCMeta):
+    """
+    For accessing memory maps.
+    This is an interface that must be implemented by sub-classes to provide platform-specific
+    support for accessing memory maps.
+    Currently the following two sub-classes are implemented:
+        1) FileAccessorWindows
+        2) FileAccessorLinux
+    """
+    @abc.abstractmethod
+    def open_mem_map(self, map_name: str, map_size: int , access: int) -> Optional[mmap.mmap]:
+        """
+        Opens an existing memory map.
+        Returns the mmap if successful, None otherwise.
+        """
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def create_mem_map(self, map_name: str, map_size: int) -> Optional[mmap.mmap]:
+        """
+        Creates a new memory map.
+        Returns the mmap if successful, None otherwise.
+        """
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def delete_mem_map(self, map_name: str, mem_map: mmap.mmap) -> bool:
+        """
+        Deletes the memory map and any backing resources associated with it.
+        If there is no memory map with the given name, then no action is performed.
+        Returns True if the memory map was successfully deleted, False otherwise.
+        """
+        raise NotImplementedError
+
+    def _verify_new_map_created(self, map_name: str, mem_map) -> bool:
+        """Checks if the first byte of the memory map is zero.
+        If it is not, this memory map already existed.
+        """
+        mem_map.seek(0)
+        byte_read = mem_map.read(1)
+        is_new_mmap = False
+        if byte_read != b'\x00':
```

Comment:
Make it a constant? With a comment explaining that memory map with 0 at the beginning doesn't exist.
BTW, we cannot get a healthy map with the first byte being 0 right?

Reply:
I think the reasoning here (which you thought of) was that for a new map, the first byte *will* be `\x00`. And if the map is already being used then it will be something other than that (like we have the header bytes which will make it non-zero) - but let me double check this.

Reply:
![image](https://user-images.githubusercontent.com/6470801/102426916-62d93f80-3fb4-11eb-972f-9c506a957193.png)
Yes, so when we create a memory map and put the header on it, the header bytes look like that ^. The first byte is non-zero so this logic seems to be fine.